src/OpenTelemetry.Exporter.Geneva/Internal/EventHeader/EventHeaderLogExporter.cs

```diff
@@ -0,0 +1,547 @@
+    private static byte GetSeverityNumber(LogLevel logLevel)
+    {
+        // Maps the Ilogger LogLevel to OpenTelemetry logging level.
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#mapping-of-severitynumber
+        // TODO: for improving perf simply do ((int)loglevel * 4) + 1
+        // or ((int)logLevel << 2) + 1
+        return logLevel switch
+        {
+            LogLevel.Trace => 1,
+            LogLevel.Debug => 5,
+            LogLevel.Information => 9,
+            LogLevel.Warning => 13,
+            LogLevel.Error => 17,
+            LogLevel.Critical => 21,
+
+            // we reach default only for LogLevel.None
+            // but that is filtered out anyway.
+            // should we throw here then?
+            LogLevel.None => 1,
+            _ => 1,
+        };
+    }
+
+    // This method would map the logger category to a table name which only contains alphanumeric values with the following additions:
+    // Any character that is not allowed will be removed.
+    // If the resulting string is longer than 50 characters, only the first 50 characters will be taken.
+    // If the first character in the resulting string is a lower-case alphabet, it will be converted to the corresponding upper-case.
+    // If the resulting string still does not comply with Rule, the category name will not be serialized.
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static string GetSanitizedCategoryName(string categoryName)
+    {
+        var validNameLength = 0;
+        Span<char> result = stackalloc char[MaxSanitizedEventNameLength];
+
+        // Special treatment for the first character.
+        var firstChar = categoryName[0];
+        if (firstChar is >= 'A' and <= 'Z')
+        {
+            result[0] = firstChar;
+            ++validNameLength;
+        }
+        else if (firstChar is >= 'a' and <= 'z')
+        {
+            // If the first character in the resulting string is a lower-case alphabet,
+            // it will be converted to the corresponding upper-case.
+            result[0] = (char)(firstChar - 32);
+            ++validNameLength;
+        }
+        else
+        {
+            // Not a valid name.
+            return string.Empty;
+        }
+
+        for (var i = 1; i < categoryName.Length; i++)
+        {
+            if (validNameLength == MaxSanitizedEventNameLength)
+            {
+                break;
+            }
+
+            var cur = categoryName[i];
+            if (cur is (>= 'a' and <= 'z') or (>= 'A' and <= 'Z') or (>= '0' and <= '9'))
+            {
+                result[validNameLength] = cur;
+                ++validNameLength;
+            }
+        }
+
+        return result.Slice(0, validNameLength).ToString();
+    }
+
+    private static void OnProcessScopeForIndividualColumns(LogRecordScope scope, EventHeaderLogExporter state)
```

Comment:
I would have made easier to review if the common methods are copied to new file and both Tld and this exporter utilizes it.

Reply:
I've took the common fields and methods out into a new base class: `TldLogCommon`.